src/core/unit.c

```diff
@@ -2225,6 +2225,11 @@ int unit_add_dependency(Unit *u, UnitDependency d, Unit *other, bool add_referen
                 return 0;
         }
 
+        if (d == UNIT_BEFORE && other->type == UNIT_DEVICE) {
+                log_unit_warning(u, "Dependency Before=%s ignored (.device units cannot be delayed)", other->id);
+                return 0;
+        }
+
```

Comment:
Actually, it's a device (waiting) job what gets ordered (delayed) and can be useful as utilized in 4195077ab4c823c385cdc3c868a56178f8d93723.

Shouldn't we change this to a mere warning (without the ignore)? 
(Note this doesn't affect 4195077ab4c823c385cdc3c868a56178f8d93723 because the endpoint of the dependency is opposite (the direction is same).)

(I'm not familiar with these ex-post commit comments, so I'll add CC: @keszybz, @Werkov.)